tests/Functional/ControllerTest.php

```diff
@@ -0,0 +1,211 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Violines\RestBundle\Tests\Functional;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Bundle\FrameworkBundle\FrameworkBundle;
+use Symfony\Bundle\FrameworkBundle\Kernel\MicroKernelTrait;
+use Symfony\Component\Config\Loader\LoaderInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Kernel;
+use Symfony\Component\Routing\Route;
+use Symfony\Component\Routing\RouteCollectionBuilder;
+use Violines\RestBundle\Tests\Stubs\MimeTypes;
+use Violines\RestBundle\ViolinesRestBundle;
+
+/**
+ * @coversNothing
+ */
+final class ControllerTest extends TestCase
+{
+    private TestKernel $app;
+
+    protected function setUp(): void
+    {
+        $this->deleteTempDir();
+
+        $this->app = new TestKernel();
+        $this->app->boot();
+    }
+
+    protected function tearDown(): void
+    {
+        $this->deleteTempDir();
+    }
+
+    public function testReturnsOne(): void
+    {
+        $request = Request::create('/returnsOne');
+        $request->headers->set('Accept', MimeTypes::APPLICATION_JSON);
+
+        $response = $this->app->handle($request);
+
+        self::assertSame(MimeTypes::APPLICATION_JSON, $response->headers->get('Content-Type'));
+        self::assertJsonStringEqualsJsonString(<<<JSON
+            {"from":"Forest"}
+       JSON, $response->getContent());
+    }
+
+    public function testReturnsMany(): void
+    {
+        $request = Request::create('/returnsMany');
+        $request->headers->set('Accept', MimeTypes::APPLICATION_JSON);
+
+        $response = $this->app->handle($request);
+
+        self::assertSame(MimeTypes::APPLICATION_JSON, $response->headers->get('Content-Type'));
+        self::assertJsonStringEqualsJsonString(<<<JSON
+            [
+                {"from":"Jenny"},
+                {"from":"Forest"}
+            ]
+       JSON, $response->getContent());
+    }
+
+    public function testReconstitutesOneFromRequest(): void
+    {
+        $submitted = <<<JSON
+            {"from":"Jenny"}
+        JSON;
+
+        $request = Request::create('/reconstitutesOne', Request::METHOD_POST, [], [], [], [], $submitted);
```

Comment:
we could think of creating a little request builder, to achieve something like:
```
$this->requestBuilder->route('/reconstitutesOne')
    ->method(Request::METHOD_POST)
    ->content($submitted)
    ->addHeader('Content-Type', MimeTypes::APPLICATION_JSON)
    ->addHeader('Accept', MimeTypes::APPLICATION_JSON)
    ->get();
```

But this is definitely no must. Just an idea!

Reply:
I don't like the idea fo implementing a Request-Builder for a tasks that only is used in tests.
But I could think of `$this->post('/reconstitutesOne', '{...}')`

Reply:
yes and i think it could be even part of this testcase as long it is the only one using it.

I'm a fan of: make things as easy as possible and just extendable, don't foresee things!